### ISSUE TITLE
MD013: `treat_links_as_single_words` should probably also include reference links in footer

### ISSUE THREAD

**Issue by particleflux** [markdownlint/markdownlint#595]
## Describe the Enhancement:
The `treat_links_as_single_words` option allows inline links to be longer than configured line limit.

However, in a reference-style link, too long lines are still reported, e.g.:

```markdown
blah blah blah [Mounting Truecrypt volumes]

foo bar blub

[Mounting Truecrypt volumes]: https://wiki.gentoo.org/wiki/Dm-crypt#Mounting_TrueCrypt.2Ftcplay.2FVeraCrypt_volumes
```

### Impacted Rules:
MD013

## Describe the Need:
Reference style links work without having to disable line length rules

## Current Alternative
Disable the rule, or reconfigure it for longer line length
